nginxbeat.go

```diff
@@ -90,46 +95,52 @@ func (nb *Nginxbeat) Setup(b *beat.Beat) error {
 func (nb *Nginxbeat) Run(b *beat.Beat) error {
 	logp.Debug(selector, "Run nginxbeat")
 
-	var p parser.Parser
-	switch nb.format {
-	case "stub":
-		p = parser.NewStubParser()
-	case "plus":
-		p = parser.NewPlusParser()
-	}
-
-	ticker := time.NewTicker(nb.period)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-nb.done:
-			goto GotoFinish
-		case <-ticker.C:
-		}
-
-		start := time.Now()
-
-		s, err := p.Parse(*nb.url)
-		if err != nil {
-			logp.Err("Fail to read Nginx status: %v", err)
-			goto GotoNext
-		}
-		nb.events.PublishEvent(common.MapStr{
-			"@timestamp": common.Time(time.Now()),
-			"type":       "nginx",
-			"nginx":      s,
-		})
-
-	GotoNext:
-		end := time.Now()
-		duration := end.Sub(start)
-		if duration.Nanoseconds() > nb.period.Nanoseconds() {
-			logp.Warn("Ignoring tick(s) due to processing taking longer than one period")
-		}
+	for _, u := range nb.urls {
+		go func() {
+			var p parser.Parser
+			switch nb.format {
+			case "stub":
+				p = parser.NewStubParser()
+			case "plus":
+				p = parser.NewPlusParser()
+			}
+
+			ticker := time.NewTicker(nb.period)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-nb.done:
+					goto GotoFinish
```

Comment:
@ruflin so i will wait until there is one from Elastic. i will keep `goto` right now for the sake of consistency inside that func(). thx for the info :)